File: examples/java/kora-java-database-vertx/src/main/java/ru/tinkoff/kora/example/vertx/VertxMapperParameterRepository.java
package ru.tinkoff.kora.example.vertx;

import jakarta.annotation.Nullable;
import ru.tinkoff.kora.common.Mapping;
import ru.tinkoff.kora.database.common.UpdateCount;
import ru.tinkoff.kora.database.common.annotation.*;
import ru.tinkoff.kora.database.vertx.VertxRepository;
import ru.tinkoff.kora.database.vertx.mapper.parameter.VertxParameterColumnMapper;

@Repository
public interface VertxMapperParameterRepository extends VertxRepository {

    final class EntityFieldTypeParameterMapper implements VertxParameterColumnMapper<Entity.FieldType> {

        @Nullable
        @Override
        public Object apply(@Nullable Entity.FieldType fieldType) {
            return (fieldType == null)
                    ? null
                    : fieldType.code();
        }
    }

    record Entity(String id,
                  @Column("value1") int field1,
                  String value2,
                  @Nullable String value3) {

        enum FieldType {

            UNKNOWN(-10),
            ONE(1),
            TWO(2);

            private final int code;

            FieldType(int code) {
                this.code = code;
            }

            public int code() {
                return code;
            }
        }
    }

    @Query("""
            UPDATE entities
            SET value1 = :fieldType
            WHERE id = :id
            """)
    UpdateCount updateFieldType(String id,
                                @Mapping(EntityFieldTypeParameterMapper.class) Entity.FieldType fieldType);
}

File: examples/kotlin/kora-kotlin-database-vertx/src/main/kotlin/ru/tinkoff/kora/kotlin/example/vertx/VertxMapperParameterRepository.kt
package ru.tinkoff.kora.kotlin.example.vertx

import io.vertx.sqlclient.Row
import io.vertx.sqlclient.RowSet
import jakarta.annotation.Nullable
import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Mapping
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.database.common.UpdateCount
import ru.tinkoff.kora.database.common.annotation.Batch
import ru.tinkoff.kora.database.common.annotation.Column
import ru.tinkoff.kora.database.common.annotation.Query
import ru.tinkoff.kora.database.common.annotation.Repository
import ru.tinkoff.kora.database.vertx.VertxRepository
import ru.tinkoff.kora.database.vertx.mapper.parameter.VertxParameterColumnMapper
import ru.tinkoff.kora.database.vertx.mapper.result.VertxResultColumnMapper
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowMapper
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowSetMapper

@Repository
interface VertxMapperParameterRepository : VertxRepository {

    data class Entity(
        val id: String,
        @field:Column("value1") val field1: Int,
        val value2: String,
        @field:Nullable val value3: String?
    ) {
        enum class FieldType(val code: Int) {
            UNKNOWN(-10),
            ONE(1),
            TWO(2)
        }
    }

    @Query("UPDATE entities SET value1 = :fieldType WHERE id = :id")
    fun updateFieldType(
        id: String,
        @Mapping(VertxEntityFieldTypeParameterMapper::class) fieldType: Entity.FieldType
    ): UpdateCount
}

class VertxEntityFieldTypeParameterMapper :
    VertxParameterColumnMapper<VertxMapperParameterRepository.Entity.FieldType?> {
    override fun apply(fieldType: VertxMapperParameterRepository.Entity.FieldType?): Any? {
        return fieldType?.code
    }
}

